README.md

```diff
@@ -1 +1,1038 @@
+
+At the same time, AM console provides a visual and easy to use interface for managing scripts and applying them to authentication and authorization events. We will use the console to apply a script to an authentication procedure in AM.
+
+### Client-Side Scripts
+
+When used for authentication in the front channel, AM allows for creating client-side scripts that would be executed in the user agent.
+
+The use case for a client-side script is collecting information about the user agent's properties and its environment: [Geolocation](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/geolocation), IP, and whatever else that could be collected with a custom script running in a browser. Thus, the script needs to be written in JavaScript compatible with the browser.
+
+From the client side, the data collected with the script can be submitted to the server and become available to the server-side scripts involved in the same authentication procedure.
+
+### Server-Side Scripts
+
+The decision making process on user identification and access management can be aided with the server-side scripts. The server-side scripts can be written in Groovy or JavaScript; with the latter running on [Rhino](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/Rhino).
+
+The server-side scripts can accept data from the client side via a well-know variable.
+
+### Authentication Trees and Authentication Chains
+
+AM supports two basic authentication workflows: `trees` and `chains`.
+
+The latter approach allows for use of a client-side script defined directly in AM console. Thus, to observe a scripted authentication with a client-side script you could use an authentication chain configured in the a realm of your choice.
+
+> The Top Level Realm is created by default during AM installation. You can find more information about [Setting Up Realms](https://backstage.forgerock.com/docs/am/6.5/maintenance-guide/index.html#chap-maint-realms) in the docs.
+
+### Scripting in AM Console
+
+To ensure you sign in AM with administrative configuration, navigate to your AM instance using the `/console` path. For example:
+
+https://my-namespace.iam.example.com/am/console/
+
+> AM allows to use separate authentication chains for administrative and non-administrative users. The choice can be made in AM's administrative console under Realms > _Realm Name_ > Authentication > Settings > Core—by selecting desired values for Administrator Authentication Configuration and Organization Authentication Configuration inputs.
+>
+> If you don't use the `/console` path, the default _Organization Authentication Configuration_ chain will be used, regardless of whether the actual user is an administrator or not.
+>
+> This may not matter if both options point to the same authentication chain.
```

Comment:
This section (from "To ensure you sign in AM with administrative configuration" to this line) strikes me as unnecessary for this article. We can assume the read knows how to login to the admin console.